modules/Bio/EnsEMBL/Compara/RunnableDB/LoadMLSSids.pm

```diff
@@ -0,0 +1,178 @@
+ - If method_type is EPO, fetch the EPO Extended MLSS id and its related GERP
+   MLSS ids and return them under "ext_mlss_id", "ce_mlss_id" and "cs_mlss_id"
+   keys, respectively
+ - If method_type is EPO_EXTENDED or PECAN, fetch the GERP MLSS ids and return
+   them under "ce_mlss_id" and "cs_mlss_id" keys
+By default, do not fetch them.
+
+=item branch_code
+
+Optional. Flow the MLSS id(s) into the given branch instead of saving them as
+pipeline-wide parameters. By default, save the MLSS id(s) as pipeline-wide
+parameters.
+
+=back
+
+=head1 EXAMPLES
+
+    standaloneJob.pl Bio::EnsEMBL::Compara::RunnableDB::LoadMLSSids \
+        --reg_conf $COMPARA_REG_PATH --master_db compara_master \
+        --method_type EPO --species_set_name mammals --release $CURR_ENSEMBL_RELEASE
+
+    standaloneJob.pl Bio::EnsEMBL::Compara::RunnableDB::LoadMLSSids \
+        --master_db $(cp1 details url ensembl_compara_master) \
+        --method_type EPO --species_set_name sauropsids --release 101 \
+        --add_prev_mlss 1 --add_sister_mlsss 1 --branch_code 2
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::LoadMLSSids;
+
+use warnings;
+use strict;
+
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+
+sub param_defaults {
+    my $self = shift;
+
+    return {
+        %{$self->SUPER::param_defaults},
+
+        'add_prev_mlss'     => 0,
+        'add_sister_mlsss'  => 0,
+        'branch_code'       => undef,
+    }
+}
+
+
+sub fetch_input {
+    my $self = shift;
+    my $master_dba = $self->get_cached_compara_dba('master_db');
+    my $release = $self->param_required('release');
+    my $method_type = $self->param_required('method_type');
+    my $species_set_name = $self->param_required('species_set_name');
+
+    my %mlss_ids;
+    my $mlss_adaptor = $master_dba->get_MethodLinkSpeciesSetAdaptor;
+    # First, fetch all the MLSSs that match the given method type and species set name
+    my @mlsss = grep { $_->species_set->name eq $species_set_name }
+        @{ $mlss_adaptor->fetch_all_by_method_link_type($method_type) };
+    $self->throw("No MLSSs found for method '$method_type' and species set '$species_set_name'") unless @mlsss;
+
+    # Fetch the "current" MLSS id (for the given release number)
+    my @curr_mlss = grep { $_->is_in_release($release) } @mlsss;
+    $self->throw("No MLSS found for method '$method_type' and species set '$species_set_name' in release $release") unless @curr_mlss;
+    my $mlss = $curr_mlss[0];
+    $mlss_ids{mlss_id} = $mlss->dbID;
+
+    if ( $self->param('add_prev_mlss') ) {
+        # Fetch the previous MLSS id
+        my $prev_release = $mlss->first_release - 1;
+        my @prev_mlss = grep { $_->last_release == $prev_release } @mlsss;
+        $self->throw(sprintf("No previous MLSS found for MLSS '%s' (%s)", $mlss->name, $mlss->dbID)) unless @prev_mlss;
+        $mlss_ids{prev_mlss_id} = $prev_mlss[0]->dbID;
+    }
```

Comment:
Do you have an example where this would be used ? Otherwise I can't tell whether this is the behaviour we need

Reply:
I need it for the new update MSA pipeline, as I need to load the data from the previous MLSS to update it. Not sure if other pipelines will use it, but it is a feature I definitely need for mine.